lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -41,6 +41,9 @@ def compute_queue_time(env)
               service: configuration[:web_service_name]
             )
 
+            # Will be set later, avoid setting the trace.resource to propagate to rack span resource
+            frontend_span.resource = nil
```

Comment:
I'm curious, does this do anything? `Tracing.trace` delegates to `Tracer#trace` which appears to me to use `resource: nil` as a default.

Reply:
Yes, it makes a difference, in terms of delaying the decision of setting resource for `TraceOperation`. When a `TraceOperation` initialize without an explicit `resource`, `@resource` is only set after the invocation of method `resource`(see [here](https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/tracing/trace_operation.rb#L134-L136
)). 

My train of thoughts of this implementation is to allow the downstream tracing behaviour determine the rack resource by setting the `trace.resource`, hence I want to avoid upstream tracing for proxy span writing to `resource` just like the rack span.

Reply:
I see, thanks for clarifying this 🙇

Reply:
After discussion, I've change the implementation. Instead of setting proxy span's resource as `nil`, implement `TraceOperation#resource_override?` to answer whether `trace.resource` has been explicitly set, to distinguish the delegation in `TraceOperation#resource`

The priority of rack span resource is 

1. User overrides `span.resource`
2. Configuration 
4. Nested App override trace.resource, eq `UsersController#show`
5. Fallback with verb + status, eq `GET 200`

https://github.com/DataDog/dd-trace-rb/pull/2180/files#diff-5934ebefd042f70e1c83157584faa0f1d97f1777aa9621d089e0dc3273108c79R139-R151